### PR TITLE
[ENH]  Allows 1 mismatch in `RotationForest` tests

### PR DESCRIPTION
**Pull request by akshathmangudi**
#### Reference Issues/PRs
FIXES: resolves issue #8620

This PR fixes CI failure on ARM architecture, specifically `ubuntu-22.04-arm` runners for RotationForest tests. The issue occurs due to floating-point precision differences between ARM and x86 architectures, causing 1 out of 300 labels to be wrong. 

#### What does this implement/fix? Explain your changes.

*Changes made:*
- Added platform detection for ARM (aarch64) architecture using `platform.machine()`
- Implemented conditional test skipping for RotationForest specifically on ARM. 
- Added a tolerance level of 1 mismatch when the specific error pattern is detected. 

No new dependencies are introduced. Uses only the standard `platform` module. 

#### What should a reviewer concentrate their feedback on?
- Check whether the platform detection logic for ARM architecture. 
- Error pattern matching for the specific test failure. 
- Test skipping implementation and messaging. 
- Whether this is the appropriate solution or the PR can be extended for possibly diagnosing the problem and fixing it. 

#### Did you add any tests for the change?
No new tests were added. The fix only modifies the existing test behavior to handle platform-specific differences.  

#### Any other comments?
**Exact diff change**: 

```
diff --git a/sktime/classification/sklearn/tests/test_all_classifiers.py b/sktime/classification/sklearn/tests/test_all_classifiers.py
index 41cb5bef9..64101d8e6 100644
--- a/sktime/classification/sklearn/tests/test_all_classifiers.py
+++ b/sktime/classification/sklearn/tests/test_all_classifiers.py
@@ -2,7 +2,9 @@
 
 __author__ = ["MatthewMiddlehurst"]
 
+import platform
 import pytest
+import numpy as np
 from sklearn.linear_model import LogisticRegression
 from sklearn.utils.estimator_checks import parametrize_with_checks
 
@@ -32,4 +34,8 @@ def test_sklearn_compatible_estimator(estimator, check):
```

##### For all contributions
- [X] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

File: sktime/classification/sklearn/tests/test_all_classifiers.py [hunk 1]
```diff
@@ -32,4 +34,8 @@ def test_sklearn_compatible_estimator(estimator, check):
         if not isinstance(
             estimator, ContinuousIntervalTree
         ) or "check for NaN and inf" not in str(error):
+            # Handle ARM architecture tolerance for RotationForest
+            if isinstance(estimator, RotationForest) and platform.machine() == "aarch64":
+                if "Arrays are not equal" in str(error) and "Mismatched elements: 1" in str(error):
+                    pytest.skip("Allowing 1 mismatch on ARM architecture for RotationForest")
```
Comment (fkiraly) [inline sktime/classification/sklearn/tests/test_all_classifiers.py]: This should be `return None` or `return`

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks for the fix! Minor change request.

File: sktime/classification/sklearn/tests/test_all_classifiers.py (hunk 1 above)
Comment (akshathmangudi) [reply sktime/classification/sklearn/tests/test_all_classifiers.py]: Hi, thanks for the feedback! I'll get back to you with the reworked fix.

**Comment by akshathmangudi** [comment]
Hi @fkiraly, can you please take a look now?

**Comment by fkiraly** [comment]
code formatting checks fail, see https://www.sktime.net/en/stable/developer_guide/coding_standards.html

**Comment by akshathmangudi** [comment]
Can you please check it now?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

There are already two stepouts now which makes the tests harder to read, could you perhaps move this to a separate function `_handle_test_stepouts(estimator, error)`?

**Comment by akshathmangudi** [comment]
hi, i've modified the test like you asked and we've created a new function `_handle_test_stepouts(estimator, error)`. the test-cases do pass and pre-commit checks were ran.

**Comment by akshathmangudi** [comment]
@fkiraly the checks seem to have passed, can you please review it?

File: sktime/classification/sklearn/tests/test_all_classifiers.py [hunk 2]
```diff
@@ -18,6 +20,25 @@
...
+    """Handle specific test stepouts for different estimators and platforms."""
+    # ContinuousIntervalTree can handle NaN values
+    if isinstance(estimator, ContinuousIntervalTree) and "check for NaN and inf" in str(
+        error
+    ):
+        return
+
+    # Handle ARM architecture tolerance for RotationForest
+    if isinstance(estimator, RotationForest) and platform.machine() == "aarch64":
+        if "Arrays are not equal" in str(error) and "Mismatched elements: 1" in str(
```
Comment (fkiraly) [inline sktime/classification/sklearn/tests/test_all_classifiers.py]: this str could also apply to 10, 11, 100, or 199 mismatched elements, no?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great!

I noticed that the string does also allows for a number of mistmatched elements that starts with 1 but is not 1. That should be fixed.

**Review by fkiraly — APPROVED** [review]
Fixed by adding a space
